internal/state/validation.go

```diff
@@ -112,6 +113,10 @@ func validateBlock(state State, block *types.Block) error {
 	}
 
 	// Validate block Time
+	if block.Time != cmttime.Canonical(block.Time) {
```

Comment:
Is this check a sanity check?

Reply:
Yes, because of the "stripping", removing the monotonic part.